deepsensor/model/model.py

```diff
@@ -648,6 +648,597 @@ def unnormalise_pred_array(arr, **kwargs):
+                )
+
+            data_x1_index, data_x2_index = get_index(data_x1, data_x2)
+            patches_clipped = {var_name: [] for var_name in patch_preds[0].keys()}
+
+            for i, patch_pred in enumerate(patch_preds):
+                for var_name, data_array in patch_pred.items():
+                    if var_name in patch_pred:
+                        # Get row/col index values of each patch. Order depends on whether coordinate is ascending or descending.
+                        if x1_ascend:
+                            patch_x1 = (
+                                data_array.coords[orig_x1_name].min().values,
+                                data_array.coords[orig_x1_name].max().values,
+                            )
+                        else:
+                            patch_x1 = (
+                                data_array.coords[orig_x1_name].max().values,
+                                data_array.coords[orig_x1_name].min().values,
+                            )
+                        if x2_ascend:
+                            patch_x2 = (
+                                data_array.coords[orig_x2_name].min().values,
+                                data_array.coords[orig_x2_name].max().values,
+                            )
+                        else:
+                            patch_x2 = (
+                                data_array.coords[orig_x2_name].max().values,
+                                data_array.coords[orig_x2_name].min().values,
+                            )
+                        patch_x1_index, patch_x2_index = get_index(patch_x1, patch_x2)
+
+                        b_x1_min, b_x1_max = patch_overlap[0], patch_overlap[0]
+                        b_x2_min, b_x2_max = patch_overlap[1], patch_overlap[1]
+
+                        """
+                        Do not remove border for the patches along top and left of dataset and change overlap size for last patch in each row and column.
+                        
+                        At end of row (when patch_x2_index = data_x2_index), to calculate the number of pixels to remove from left hand side of patch:
+                        If x2 is ascending, subtract previous patch x2 max value from current patch x2 min value to get bespoke overlap in column pixels.
+                        To account for the clipping done to the previous patch, then subtract patch_overlap value in pixels 
+                        to get the number of pixels to remove from left hand side of patch.
+
+                        If x2 is descending. Subtract current patch max x2 value from previous patch min x2 value to get bespoke overlap in column pixels. 
+                        To account for the clipping done to the previous patch, then subtract patch_overlap value in pixels 
+                        to get the number of pixels to remove from left hand side of patch.
+
+                        """
```

Comment:
I'm not sure I agree that this would make more sense in the docstring for the method since it's a description of the internals for the developer, though I agree that it's rather large and indented. Would you be okay with splitting this into smaller comments in line with the appropriate operations?